services/manifest-repo-export-service/pkg/repository/transformer_test.go

```diff
@@ -519,7 +467,366 @@ func TestTransformerWorksWithDb(t *testing.T) {
 			if diff := cmp.Diff(tc.ExpectedError, err, cmpopts.EquateErrors()); diff != "" {
 				t.Errorf("error mismatch (-want, +got):\n%s", diff)
 			}
+			if tc.ExpectedFile != nil {
+				for i := range tc.ExpectedFile {
+					expectedFile := tc.ExpectedFile[i]
+					updatedState := repo.State()
+					fullPath := updatedState.Filesystem.Join(updatedState.Filesystem.Root(), expectedFile.path)
+					actualFileData, err := util.ReadFile(updatedState.Filesystem, fullPath)
+					if err != nil {
+						t.Fatalf("Expected no error: %v path=%s", err, fullPath)
+					}
 
+					if !cmp.Equal(actualFileData, expectedFile.fileData) {
+						t.Fatalf("Expected '%v', got '%v'", string(expectedFile.fileData), string(actualFileData))
+					}
+					if tc.ExpectedAuthor != nil {
+						if !cmp.Equal(updatedState.Commit.Author().Name, (*tc.ExpectedAuthor)["Name"]) {
+							t.Fatalf("Expected '%v', got '%v'", (*tc.ExpectedAuthor)["Name"], updatedState.Commit.Author().Name)
+						}
+						if !cmp.Equal(updatedState.Commit.Author().Email, (*tc.ExpectedAuthor)["Email"]) {
+							t.Fatalf("Expected '%v', got '%v'", (*tc.ExpectedAuthor)["Email"], updatedState.Commit.Author().Email)
+						}
+					}
+				}
+			}
+			if tc.ExpectedDeletedFile != "" {
+				updatedState := repo.State()
+				fullPath := updatedState.Filesystem.Join(updatedState.Filesystem.Root(), tc.ExpectedDeletedFile)
+				_, err := util.ReadFile(updatedState.Filesystem, fullPath)
+				if err == nil {
+					t.Fatalf("Expected file to be deleted but it exists: %s", fullPath)
+				}
+			}
+		})
+	}
+}
+func verifyContent(fs billy.Filesystem, required []*FilenameAndData) error {
+	for _, contentRequirement := range required {
+		if data, err := util.ReadFile(fs, contentRequirement.path); err != nil {
+			return fmt.Errorf("error while opening file %s, error: %w", contentRequirement.path, err)
+		} else if string(data) != string(contentRequirement.fileData) {
```

Comment:
```suggestion
		}
		if string(data) != string(contentRequirement.fileData) {
```
`else` right after a `return` does not do anything